### PR TITLE
Add missing index name to search slow log.

### LINKED ISSUE #17025: Slowlogs no longer include index name
**Elasticsearch version**: 2.2.0

**JVM version**: 1.8.0_40

**OS version**: OSX and Ubuntu 15.04

**Description of the problem including expected versus actual behavior**:
Unless I'm completely and utterly blind (which is a distinct possibility), it seems that both indexing and query slowlogs no longer include the name of the index that the operation was running against. This makes them almost entirely useless for us (since we have several hundred indices in our cluster).

**Steps to reproduce**:
 1. Generate some queries / indexing requests that triggers a slowlog.
 2. Look at the slowlog
 3. Scratch your head and wonder why the `type` is present, but not the `index`.

**Provide logs (if relevant)**:

An example slowlog from 2.2.0:

```
[2016-03-09 16:26:15,225][TRACE][index.indexing.slowlog.index] took[4.4ms], took_millis[4], type[test], id[AVNaDew0JKS5u4ProrcX], routing[] , source[{"name":"vlucas/phpdotenv","version":"1.0.3","type":"library","description":"Loads environment variables from `.env` to `getenv()`, `$_ENV` and `$_SERVER` automagically.","keywords":["env","dotenv","environment"],"homepage":"http://github.com/vlucas/phpdotenv","license":"BSD","authors":[{"name":"Vance Lucas","email":"vance@vancelucas.com","homepage":"http://www.vancelucas.com"}],"require":{"php":">=5.3.2"},"require-dev":{"phpunit/phpunit":"*"},"autoload":{"psr-0":{"Dotenv":"src/"}}}]
```

An example slowlog from 1.7.5:

```
[2016-03-09 16:31:42,346][WARN ][index.indexing.slowlog.index] [Prowler] [slowtest][1] took[95.7ms], took_millis[95], type[test], id[AVNaEumoVIxyp_lAGADK], routing[], source[{"name":"vlucas/phpdotenv","version":"1.0.3","type":"library","description":"Loads environment variables from `.env` to `getenv()`, `$_ENV` and `$_SERVER` automagically.","keywords":["env","dotenv","environment"],"homepage":"http://github.com/vlucas/phpdotenv","license":"BSD","authors":[{"name":"Vance Lucas","email":"vance@vancelucas.com","homepage":"http://www.vancelucas.com"}],"require":{"php":">=5.3.2"},"require-dev":{"phpunit/phpunit":"*"},"autoload":{"psr-0":{"Dotenv":"src/"}}}]
```

### PR DESCRIPTION
**Pull request by jimczi**
This commits adds the index name as part of the logging message.
Closes #17025

### CODE REVIEW

**Comment by ywelsch** [comment]
LGTM

**Comment by jimferenczi** [comment]
Thanks @ywelsch !

File: core/src/main/java/org/elasticsearch/index/SearchSlowLog.java [hunk 1]
```diff
@@ -120,43 +122,46 @@ private void setLevel(SlowLogLevel level) {
...
 
-    private static class SlowLogSearchContextPrinter {
+    static final class SlowLogSearchContextPrinter {
         private final SearchContext context;
+        private final Index index;
         private final long tookInNanos;
         private final boolean reformat;
 
-        public SlowLogSearchContextPrinter(SearchContext context, long tookInNanos, boolean reformat) {
+        public SlowLogSearchContextPrinter(Index index, SearchContext context, long tookInNanos, boolean reformat) {
```
Comment (markharwood) [inline core/src/main/java/org/elasticsearch/index/SearchSlowLog.java]: Just checking - doesn't SearchContext already give you access to index via searchContext.indexShard().shardId()?
I haven't tried running it but looking at SearchContext code it looks like there's already a path to this info?

**Comment by jimferenczi** [comment]
@markharwood thanks, I've missed it. Though I had to use the SearchContext for 2.x because the IndexSettings is not available in those branches. I'll update the master branch, thanks.

**Pull request merged**